crates/driver/src/domain/quote.rs

```diff
@@ -148,16 +132,66 @@ impl Order {
         match self.side {
             order::Side::Sell => eth::Asset {
                 amount: self.amount.into(),
-                token: self.sell_token,
+                token: self.tokens.sell,
             },
             order::Side::Buy => eth::Asset {
                 amount: eth::U256::max_value(),
-                token: self.sell_token,
+                token: self.tokens.sell,
             },
         }
     }
 }
 
+/// The deadline for computing a quote for an order.
+#[derive(Clone, Copy, Debug, Default)]
+pub struct Deadline(chrono::DateTime<chrono::Utc>);
```

Comment:
I like that you moved this down here :+1:

In general I try to also pay attention to this and make it so that the most "important" types come first, and the less important ones after.